From 80067ee2115498fd5225c5dbd8628332e1867846 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <viswanat@qti.qualcomm.com>
Date: Tue, 3 Mar 2026 12:27:10 +0530
Subject: [PATCH] drivers: pas: Rename qcom_pas_metadata to qcom_pas_context

Latest patchset of PAS TEE framework renames struct qcom_pas_metadata to
struct qcom_pas_context.

Handle this change in all the required files.

Change-Id: Ie0f699418d7af6315a869709121264fd3bd2b538
Signed-off-by: Vignesh Viswanathan <viswanat@qti.qualcomm.com>
---
 drivers/firmware/Kconfig            | 3 ++-
 drivers/firmware/qcom_scm.c         | 4 ++--
 drivers/remoteproc/qcom_q6v5_pas.c  | 4 ++--
 drivers/soc/qcom/mdt_loader.c       | 2 +-
 include/linux/soc/qcom/mdt_loader.h | 6 +++---
 5 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/drivers/firmware/Kconfig b/drivers/firmware/Kconfig
index 80449f1c0e21..3b18c33c7bdd 100644
--- a/drivers/firmware/Kconfig
+++ b/drivers/firmware/Kconfig
@@ -220,9 +220,10 @@ config QCOM_PAS
 	  TEE bus based PAS service implementation.
 
 config QCOM_PAS_TEE
-	tristate "QCOM PAS framework provided by TEE"
+	tristate
 	depends on TEE
 	select QCOM_PAS
+	default m
 	help
 	  Enable the generic Peripheral Authentication Service (PAS) provided
 	  by the firmware TEE implementation as the backend.
diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index aaa1ab07f4ee..8eedf6dfe982 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -708,7 +708,7 @@ static void qcom_scm_set_download_mode(bool enable)
  */
 static int qcom_scm_pas_init_image(struct device *dev, u32 peripheral,
 				   const void *metadata, size_t size,
-				   struct qcom_pas_metadata *ctx)
+				   struct qcom_pas_context *ctx)
 {
 	dma_addr_t mdata_phys;
 	void *mdata_buf;
@@ -774,7 +774,7 @@ static int qcom_scm_pas_init_image(struct device *dev, u32 peripheral,
  * @ctx:	metadata context
  */
 static void qcom_scm_pas_metadata_release(struct device *dev,
-					  struct qcom_pas_metadata *ctx)
+					  struct qcom_pas_context *ctx)
 {
 	if (!ctx->ptr)
 		return;
diff --git a/drivers/remoteproc/qcom_q6v5_pas.c b/drivers/remoteproc/qcom_q6v5_pas.c
index d0189aca8634..fb8c8062d913 100644
--- a/drivers/remoteproc/qcom_q6v5_pas.c
+++ b/drivers/remoteproc/qcom_q6v5_pas.c
@@ -103,8 +103,8 @@ struct qcom_adsp {
 	struct qcom_rproc_ssr ssr_subdev;
 	struct qcom_sysmon *sysmon;
 
-	struct qcom_pas_metadata pas_metadata;
-	struct qcom_pas_metadata dtb_pas_metadata;
+	struct qcom_pas_context pas_metadata;
+	struct qcom_pas_context dtb_pas_metadata;
 };
 
 static void adsp_segment_dump(struct rproc *rproc, struct rproc_dump_segment *segment,
diff --git a/drivers/soc/qcom/mdt_loader.c b/drivers/soc/qcom/mdt_loader.c
index b046ecede2f0..5f563d6d8d34 100644
--- a/drivers/soc/qcom/mdt_loader.c
+++ b/drivers/soc/qcom/mdt_loader.c
@@ -313,7 +313,7 @@ EXPORT_SYMBOL_GPL(qcom_mdt_read_metadata);
  */
 int qcom_mdt_pas_init(struct device *dev, const struct firmware *fw,
 		      const char *fw_name, int pas_id, phys_addr_t mem_phys,
-		      struct qcom_pas_metadata *ctx)
+		      struct qcom_pas_context *ctx)
 {
 	const struct elf32_phdr *phdrs;
 	const struct elf32_phdr *phdr;
diff --git a/include/linux/soc/qcom/mdt_loader.h b/include/linux/soc/qcom/mdt_loader.h
index 3bf9c3c1cfba..a8231eb7442f 100644
--- a/include/linux/soc/qcom/mdt_loader.h
+++ b/include/linux/soc/qcom/mdt_loader.h
@@ -15,7 +15,7 @@
 
 struct device;
 struct firmware;
-struct qcom_pas_metadata;
+struct qcom_pas_context;
 
 #if IS_ENABLED(CONFIG_QCOM_MDT_LOADER)
 
@@ -23,7 +23,7 @@ ssize_t qcom_mdt_get_size(const struct firmware *fw);
 ssize_t qcom_mdt_get_file_size(const struct firmware *fw);
 int qcom_mdt_pas_init(struct device *dev, const struct firmware *fw,
 		      const char *fw_name, int pas_id, phys_addr_t mem_phys,
-		      struct qcom_pas_metadata *pas_metadata_ctx);
+		      struct qcom_pas_context *pas_metadata_ctx);
 int qcom_mdt_load(struct device *dev, const struct firmware *fw,
 		  const char *fw_name, int pas_id, void *mem_region,
 		  phys_addr_t mem_phys, size_t mem_size,
@@ -49,7 +49,7 @@ static inline ssize_t qcom_mdt_get_size(const struct firmware *fw)
 
 static inline int qcom_mdt_pas_init(struct device *dev, const struct firmware *fw,
 				    const char *fw_name, int pas_id, phys_addr_t mem_phys,
-				    struct qcom_pas_metadata *pas_metadata_ctx)
+				    struct qcom_pas_context *pas_metadata_ctx)
 {
 	return -ENODEV;
 }
-- 
2.34.1

